readme.md

```diff
@@ -92,36 +155,59 @@ The `pkcs1-reader-sun` does not include any extra dependencies, but assumes the
 available as part of most popular JVM implementations, including the Oracle and OpenJDK JVMs for
 Java 7 and Java 8.
 
-### Add Dependency Using Gradle
+## error-handling
+Provides utilities for relaying Java exceptions across JVM boundaries by serializing exceptions as JSON POJOs.
 
-```groovy
-repositories {
-  jcenter()
-}
-dependencies {
-  compile "com.palantir.remoting:http-clients:$version"
-  compile "com.palantir.remoting:retrofit-clients:$version"
-  compile "com.palantir.remoting:http-servers:$version"
-  compile "com.palantir.remoting:ssl-config:$version"
-  compile "com.palantir.remoting:error-handling:$version"
-  // support for PEM key store type using Bouncy Castle libraries
-  // compile "com.palantir.remoting:pkcs1-reader-bouncy-castle:$version"
-  // support for PEM key store type using Sun libraries
-  // compile "com.palantir.remoting:pkcs1-reader-sun:$version"
-}
-```
 
+# API Contract
+
+http-remoting makes the following opinionated customizations to the standard Dropwizard/Feign/Retrofit behavior.
+
+#### Object serialization/deserialization
+
+All parameters and return values of `application/json` endpoints are serialized/deserialized to/from JSON using a
+Jackson `ObjectMapper` with `GuavaModule` and `Jdk7Module`. Servers must not expose parameters or return values that
+cannot be handled by this object mapper.
+
+#### Error propagation
+The `DropwizardServers#configure` routine installs exception mappers for `IllegalArgumentException`,
+`NoContentException`, `RuntimeException` and `WebApplicationException`. The exception mapper sets the response media
+type to `application/json` and returns as response body a JSON representation of a `SerializableError` capturing the
+message, exception name, and optionally stacktrace of the exception.
+
+Both JaxRsClient and Retrofit2Client intercept non-successful HTTP responses and throw a `RemoteException` wrapping the
+deserialized server-side `SerializableError`. The error name of the `RemoteException` is defined by the service API and
+clients should switch&dispatch based on the error name.
+
+#### Optional return values
```

Comment:
it's down there (although the sentence wasn't complete ...): 
```
 202 - a non-empty return value, then the HTTP response has error code 200 and the body carries the deserialized `T` object
 203   directly, rather than a deserialized `Optional<T>` object.
```